airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -2630,6 +2630,11 @@ def run_query(self, *args, **kwargs) -> str:
             stacklevel=3,
         )
         return self.hook.run_query(*args, **kwargs)
+    
+    def get_records(self, sql, parameters=None):
+        if self.location is None:
+            raise Exception("Need to specify location when instantiating BigQueryHook, otherwise it would result in Job Not Found error!")
```

Comment:
Instead of failing on runtime, can we instead detect this in `__init__` and fail the entire DAG when `location` is invalid? Or is `location=None` a valid value for certain use cases?

Reply:
yea `location=None` works for other function and seems to me that it only fails when invoking `get_records()` ...
btw I might need to postpone this PR, will try to finish it by the end of Nov
will ping @uranusjr for help 😂

Reply:
OK this sounds like a good reason.

Could you change this to raise `AirflowException` (or a subclass `BigQueryLocationUnset`) instead? This pattern is more common in Airflow than raising a barebone `Exception`.

I’d do

```python
raise BigQueryLocationUnset("Parameter 'location' is required to fetch records with BigQuery hook")
```

Reply:
Sure, will do. Thx for the inputs 🎉